### PR TITLE
Improve format devexp

### PR DESCRIPTION
**Pull request by oumar-fall**
- Add `createFormat` function to create formatters with default options
- Add `createOf` method on formatters to create a standalone formatting function

### CODE REVIEW

File: packages/morpho-ts/README.md [hunk 1]
```diff
@@ -47,27 +47,45 @@ Here are all the available formatters:
 - [**Short Formatter**](#short-formatter): Formats the value in a short notation with units (e.g., `k`, `M`).
 - [**Percent Formatter**](#percent-formatter): Formats the value as a percentage.
 
-### Usage
+#### Usage
```
Comment (Rubilmax) [inline packages/morpho-ts/README.md:50]: I know it's not indented correctly, but h4 is not that visible hence why I kept h3
Wdyt?

File: packages/morpho-ts/README.md [hunk 2]
```diff
@@ -168,6 +186,29 @@ format.percent.digits(1).sign().of(0.123456); // "+12.3%"
 
 ---
 
+#### Create custom formatters
```
Comment (Rubilmax) [inline packages/morpho-ts/README.md:189]:
```suggestion
### Custom formatter factory
```

**Review by Rubilmax — COMMENTED** [review]
Nice one, just have remarks on the README

File: packages/morpho-ts/README.md (hunk 1 above)
Comment (oumar-fall) [reply packages/morpho-ts/README.md:50]: I don't care much tbh

**Review by Rubilmax — APPROVED** [review]

**Pull request merged**